pkg/controller/machine_util.go

```diff
@@ -266,3 +268,121 @@ func nodeConditionsHaveChanged(machineConditions []v1.NodeCondition, nodeConditi
 
 	return false
 }
+
+// syncMachineNodeTemplate syncs nodeTemplates between machine and corresponding node-object.
+// It ensures, that any nodeTemplate element available on Machine should be available on node-object.
+// Although there could be more elements already available on node-object which will not be touched.
+func (c *controller) syncMachineNodeTemplates(machine *v1alpha1.Machine) error {
+	if machine.Status.Node == "" {
```

Comment:
If a machine object doesn't have a node object associated with it due to some error while creation. Could it be a possibility that the object doesn't get deleted? as labels are tried to sync and it fails due to no node object backing it? or maybe I didn't get it.

Reply:
Yes, this is a very good catch, thanks.
Basically if, for any reason, node-object is available and `Status.Node` has not been updated then the controller can get stuck into the loop.